crates/sui-framework/sources/Governance/ValidatorSet.move

```diff
@@ -327,22 +343,69 @@ module Sui::ValidatorSet {
         }
     }
 
+    /// Given the current list of active validators, the total stake and total reward,
+    /// calculate the amount of reward each validator should get.
+    /// Returns the amount of reward for each validator, as well as a remaining reward
+    /// due to integer division loss.
+    fun compute_reward_distribution(
+        validators: &vector<Validator>,
+        total_stake: u64,
+        total_reward: u64,
+    ): vector<u64> {
+        let results = Vector::empty();
+        let length = Vector::length(validators);
+        let i = 0;
+        while (i < length) {
+            let validator = Vector::borrow(validators, i);
+            // Integer divisions will truncate the results. Because of this, we expect that at the end
+            // there will be some reward remaining in `total_reward`.
+            // Use u128 to avoid multiplication overflow.
+            let stake_amount: u128 = (Validator::stake_amount(validator) as u128);
+            let reward_amount = stake_amount * (total_reward as u128) / (total_stake as u128);
+            Vector::push_back(&mut results, (reward_amount as u64));
+            i = i + 1;
+        };
+        results
+    }
+
     // TODO: Allow reward compunding for delegators.
-    fun distribute_reward(validators: &mut vector<Validator>, total_stake: u64, reward: &mut Balance<SUI>) {
-        let total_reward_amount = Balance::value(reward);
+    fun distribute_reward(validators: &mut vector<Validator>, rewards: &vector<u64>, reward: &mut Balance<SUI>) {
         let length = Vector::length(validators);
         let i = 0;
         while (i < length) {
             let validator = Vector::borrow_mut(validators, i);
-            // Integer divisions will truncate the results. Because of this, we expect that at the end
-            // there will be some reward remaining in `reward`.
-            let reward_amount = Validator::stake_amount(validator) * total_reward_amount / total_stake;
+            let reward_amount = *Vector::borrow(rewards, i);
             let reward = Balance::split(reward, reward_amount);
-            Validator::deposit_reward(validator, reward);
+            // Because reward goes to pending stake, it's the same as calling `request_add_stake`.
+            Validator::request_add_stake(validator, reward);
             i = i + 1;
         }
     }
 
+    /// Upon any change to the validator set, derive and update the metadata of the validators for the new epoch.
+    fun derive_next_epoch_validators(self: &ValidatorSet): vector<ValidatorMetadata> {
```

Comment:
Is this the place where we have some minimum stake by % to becomes a validator?

Reply:
Correct!